From: NeilBrown <neilb@suse.de>
Date: Mon, 31 Jan 2011 11:57:42 +1100
Subject: [PATCH] md: revert change to raid_disks on failure.
Git-commit: de171cb9a52598cc023adceafc6c166112401386
Patch-mainline: v2.6.38-rc7

If we try to update_raid_disks and it fails, we should put
'delta_disks' back to zero.  This is important because some code,
such as slot_store, assumes that delta_disks has been validated.

Signed-off-by: NeilBrown <neilb@suse.de>
---
 drivers/md/md.c |    2 ++
 1 file changed, 2 insertions(+)

--- linux-2.6.32-SLE11-SP1.orig/drivers/md/md.c
+++ linux-2.6.32-SLE11-SP1/drivers/md/md.c
@@ -5546,6 +5546,8 @@ static int update_raid_disks(mddev_t *md
 	mddev->delta_disks = raid_disks - mddev->raid_disks;
 
 	rv = mddev->pers->check_reshape(mddev);
+	if (rv < 0)
+		mddev->delta_disks = 0;
 	return rv;
 }
 
